File: app/src/main/java/com/ivanmagda/habito/barchart/formatters/HabitoBaseIAxisValueFormatter.java
package com.ivanmagda.habito.barchart.formatters;

import com.github.mikephil.charting.components.AxisBase;
import com.github.mikephil.charting.formatter.IAxisValueFormatter;

public abstract class HabitoBaseIAxisValueFormatter implements IAxisValueFormatter {

    @Override
    public String getFormattedValue(float value, AxisBase axis) {
        return String.valueOf(value);
    }

    public abstract long getDateForValue(float value);

}

File: app/src/main/java/com/ivanmagda/habito/barchart/formatters/HabitoBaseIAxisValueFormatter.kt
package com.ivanmagda.habito.barchart.formatters

import com.github.mikephil.charting.components.AxisBase
import com.github.mikephil.charting.formatter.IAxisValueFormatter

abstract class HabitoBaseIAxisValueFormatter : IAxisValueFormatter {

    override fun getFormattedValue(value: Float, axis: AxisBase): String {
        return value.toString()
    }

    abstract fun getDateForValue(value: Float): Long
}
